lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,386 @@
+"use strict";
+
+const EMIT_DATA = "data", EMIT_ERR = "error";
+
+module.exports = (backendConfig, configValidation, crypto, EventEmitter, mysql, timers) => {
+    /**
+     * Updates heartbeatDate of specific record, logs an error if there is one,
+     * and calls callback that is passed in. This will continue until
+     * pausePolling is called.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     * @param {Function} callback
+     */
+    function heartbeat(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`UPDATE ??
+            SET heartbeatDate = NOW()
+            WHERE hash = ?
+            AND isProcessing = true
+            AND owner = ?;`,
+            [
+                ddqBackendInstance.config.table,
+                recordId,
+                ddqBackendInstance.owner
+            ],
+            (err) => {
+                if (err) {
+                    console.error(err);
+                }
+
+                callback(err);
+            }
+        );
+    }
+
+
+    /**
+     * Polls for a random record with the minimum delay defined by the
+     * config.pollDelayMs.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {Function} callback
+     */
+    function poll(ddqBackendInstance, callback) {
+        /**
+         * Polls for a random record in the database and calls the callback
+         * provided to it. This will continue until pausePolling is called.
+         */
+        function findData() {
+            ddqBackendInstance.connection.query(`SELECT hash
+                FROM ??
+                WHERE isProcessing = false
+                AND topics IN (?)
+                OR topics IS NULL
+                ORDER BY RAND()
+                LIMIT 1;`,
+                [
+                    ddqBackendInstance.config.table,
+                    ddqBackendInstance.config.topics
+                ],
+                callback
```

Comment:
From what I understand, this callback is not the one you intended.  More on this in the actual class.